samza-core/src/main/java/org/apache/samza/storage/TaskStorageCommitManager.java

```diff
@@ -139,18 +132,16 @@ public void init() {
   }
 
   /**
-   * // TODO HIGH dchen fix method name and documentation. local file system isn't very relevant here (too general).
-   * // Maybe rename to "writeCheckpointToStoreDirs" or something and update docs.
    * Writes the {@link Checkpoint} returned by {@link #commit(CheckpointId)}
    * locally to the file system on disk if the checkpoint passed in is an instance of {@link CheckpointV2},
    * otherwise if it is an instance of {@link CheckpointV1} persists the Kafka changelog ssp-offsets only.
    *
    * Note: The assumption is that this method will be invoked once for each {@link Checkpoint} version that the
-   * task needs to write.
+   * task needs to write, once per checkpoint version for backwards compatibility.
```

Comment:
"needs to write, as determined by {@link TaskConfig#getCheckpointVersionsToWrite} (or whatever the method name is). This is required for upgrade and rollback compatibility."